fhttp/http_client.go

```diff
@@ -709,13 +719,24 @@ func (c *FastClient) Fetch() (int, []byte, int) {
 	c.headerLen = 0
 	// Connect or reuse existing socket:
 	conn := c.socket
-	reuse := (conn != nil)
+
+	reuse := conn != nil
+	if c.maxConnectionReuse != 0 && c.currentConnection >= c.maxConnectionReuse {
```

Comment:
I think this may make the minimum reuse to be 2 instead of 1?

ie I should be able to say close each time, I think if I set [1, 1] wouldn't it close/redo every other?

can add a test with the socket count that if I do n: 10 and reuse [1 1] I should get exactly 10 sockets and if I put [2 2] I should get exactly 5 ... [and another boundary check is [9 9] should be... 2 sockets [10 10] should be 1

Reply:
fixed